kong/basic_auth_service_test.go

```diff
@@ -155,6 +155,58 @@ func TestBasicAuthGet(T *testing.T) {
 	require.NoError(client.Consumers.Delete(defaultCtx, consumer.ID))
 }
 
+func TestBasicAuthGetByID(t *testing.T) {
+	RunWhenDBMode(t, "postgres")
+
+	client, err := NewTestClient(nil, nil)
+	require.NoError(t, err)
+	require.NotNil(t, client)
+
+	uuid := uuid.NewString()
+	basicAuth := &BasicAuth{
+		ID:       String(uuid),
+		Username: String("my-username"),
+		Password: String("my-password"),
+	}
+
+	// consumer for the basic-auth:
+	consumer := &Consumer{
+		Username: String("foo"),
+	}
+
+	consumer, err = client.Consumers.Create(defaultCtx, consumer)
+	require.NoError(t, err)
+	require.NotNil(t, consumer)
+
+	createdBasicAuth, err := client.BasicAuths.Create(defaultCtx,
+		consumer.ID, basicAuth)
+	require.NoError(t, err)
+	require.NotNil(t, createdBasicAuth)
+
+	t.Cleanup(func() {
+		require.NoError(t, client.Consumers.Delete(defaultCtx, consumer.ID))
+	})
+
+	t.Run("successful basic-auth retrieval by ID", func(_ *testing.T) {
```

Comment:
We should use `testing.T` of this sub-test case instead of the top-level one. Same applies to every subtest.
```suggestion
	t.Run("successful basic-auth retrieval by ID", func(t *testing.T) {
```